azure/test.yml

```diff
@@ -24,33 +25,39 @@ jobs:
           vmImage: macOS-10.14
         Windows:
           vmImage: windows-2019
-  ${{ if ne('true', parameters.cross) }}:
-    variables:
+  variables:
+    ${{ if ne('true', parameters.cross) }}:
       vmImage: ubuntu-16.04
+    options: ''
   pool:
     vmImage: $(vmImage)
   steps:
   - template: _setup.yml
     parameters:
       rust: ${{ parameters.rust }}
       setup: ${{ parameters.setup }}
-  - ${{ if ne('true', parameters.single_threaded) }}:
-    - script: cargo test --all --features "${{ parameters.features }}"
-      displayName: Run tests
-      env:
-        ${{ insert }}: ${{ parameters.envs }}
+  # Cargo options go here
+  - bash: echo '##vso[task.setvariable variable=options]$(options) --all'
+  - ${{ if eq(true, parameters.features)}}:
+    - bash: echo '##vso[task.setvariable variable=options]$(options) --features "${{ parameters.features }}"'
+  # Add any raw args
+  - ${{ if eq(true, parameters.raw_opts) }}:
+    - bash: echo '##vso[task.setvariable variable=options]$(options) ${{ parameters.raw_opts }}'
+  # Add any additional conditional libtest options to this check
+  # Attempts to ensure that any raw args passed do not interfere
+  - ${{ if and(or(eq('true', parameters.single_threaded), eq('true', parameters.test_ignored)), not(endsWith(parameters.raw_opts, '--'))) }}:
```

Comment:
I don't think you need the whole `or()` part of this -- it's fine to add a trailing `--` :)